test/extended/util/disruption/controlplane/controlplane.go

```diff
@@ -162,3 +163,43 @@ func allClusterVersionsAreGTE(version semver.Version, config *rest.Config) (bool
 	}
 	return true, nil
 }
+
+type ResiliencyTest struct{}
+
+func (t ResiliencyTest) Name() string { return "resiliency-test" }
+func (t ResiliencyTest) DisplayName() string {
+	return "resilient test"
+}
+
+// Setup does nothing
+func (t *ResiliencyTest) Setup(f *framework.Framework) {
+}
+
+// Test runs a connectivity check to the core APIs.
+func (t *ResiliencyTest) Test(f *framework.Framework, done <-chan struct{}, upgrade upgrades.UpgradeType) {
+	config, err := framework.LoadConfig()
+	framework.ExpectNoError(err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	m := monitor.NewMonitorWithInterval(time.Second)
+	err = monitor.StartKubeAPIMonitoringWithNewConnections(ctx, m, config, 15*time.Second)
+	framework.ExpectNoError(err, "unable to monitor API")
+
+	start := time.Now()
+	m.StartSampling(ctx)
+
+	// wait to ensure API is still up after the test ends
+	<-done
+	time.Sleep(5 * 60 * time.Second)
+	cancel()
+	end := time.Now()
+
+	// TODO: validate API is down for a gracefull amount of time, and validate other operators are not affected
+	toleratedDisruption := 0.08
+	events := m.Intervals(time.Time{}, time.Time{})
+	disruption.ExpectNoDisruption(f, toleratedDisruption, end.Sub(start), events, "API was unreachable during disruption (AWS has a known issue: https://bugzilla.redhat.com/show_bug.cgi?id=1943804)")
```

Comment:
You planned to call `ExpectDisruption` instead, right?

Reply:
ignore for now

Reply:
I'll have to use other validation from those two options I think